src/EnergyPlus/HVACUnitarySystem.cc

```diff
@@ -7141,71 +7706,109 @@ namespace HVACUnitarySystem {
         if (UnitarySystem(UnitarySysNum).CoolingCoilUpstream) {
 
             if (UnitarySystem(UnitarySysNum).CoolCoilExists) {
-                UpdateUnitarySystemControl(UnitarySysNum, AirLoopNum, UnitarySystem(UnitarySysNum).CoolCoilOutletNodeNum,
-                                           UnitarySystem(UnitarySysNum).SystemCoolControlNodeNum, OnOffAirFlowRatio, FirstHVACIteration,
+                UpdateUnitarySystemControl(UnitarySysNum,
+                                           AirLoopNum,
+                                           UnitarySystem(UnitarySysNum).CoolCoilOutletNodeNum,
+                                           UnitarySystem(UnitarySysNum).SystemCoolControlNodeNum,
+                                           OnOffAirFlowRatio,
+                                           FirstHVACIteration,
                                            OAUCoilOutTemp);
                 ControlCoolingSystemToSP(UnitarySysNum, AirLoopNum, FirstHVACIteration, HXUnitOn, CompOn);
                 PartLoadRatio = UnitarySystem(UnitarySysNum).CoolingPartLoadFrac;
                 CompOn = 0;
-                if (PartLoadRatio > 0.0) CompOn = 1;
-                CalcUnitaryCoolingSystem(UnitarySysNum, AirLoopNum, FirstHVACIteration, PartLoadRatio, CompOn, OnOffAirFlowRatio, CoilCoolHeatRat,
-                                         HXUnitOn);
+                if (PartLoadRatio > 0.0) {
+                    CompOn = 1;
```

Comment:
I'm saving the operating mode when SP based coils are called so that I have a way to set flow in Init for SP based control.